clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java

```diff
@@ -525,16 +526,37 @@ public void testHeartbeatState() {
 
     @Test
     public void testPollTimerExpiration() {
-        heartbeatRequestManager = createHeartbeatRequestManager();
+        coordinatorRequestManager = mock(CoordinatorRequestManager.class);
+        membershipManager = mock(MembershipManager.class);
+        heartbeatState = mock(HeartbeatRequestManager.HeartbeatState.class);
+        heartbeatRequestState = spy(new HeartbeatRequestManager.HeartbeatRequestState(
```

Comment:
Why is there still this spy?

Reply:
If it is too much work to change, let's change it later.

Reply:
it is there to verify the reset call and we also need a real object to ensure the manager can send a request.  If we mock it, then we have to mock the canSendRequest call, then it is pointless to verify the request returned from poll because it will always return one.